test/forms/traderDetails/TirIdNumberFormProviderSpec.scala

```diff
@@ -26,11 +25,10 @@ class TirIdNumberFormProviderSpec extends StringFieldBehaviours with FieldBehavi
 
   private val prefix = Gen.alphaNumStr.sample.value
 
-  private val requiredKey      = s"$prefix.error.required"
-  private val maxLengthKey     = s"$prefix.error.maxLength"
-  private val invalidFormatKey = s"$prefix.error.invalidFormat"
-
-  private val form = new TirIdNumberFormProvider()(prefix)
+  private val requiredKey          = s"$prefix.error.required"
+  private val invalidFormatKey     = s"$prefix.error.invalidFormat"
+  private val maxTirIdNumberLength = 17
```

Comment:
There's a `stringsThatMatchRegex` method that may be of use here? Rather than using stringsWithMaxLength when we're getting rid of the length constraint